validator/db/filesystem/proposer_settings.go

```diff
@@ -0,0 +1,237 @@
+package filesystem
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+
+	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
+	validatorServiceConfig "github.com/prysmaticlabs/prysm/v4/config/validator/service"
+	validatorpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1/validator-client"
+)
+
+// ErrNoProposerSettingsFound is an error thrown when no settings are found.
+var ErrNoProposerSettingsFound = errors.New("no proposer settings found in bucket")
+
+// ProposerSettings returns the proposer settings.
+func (s *Store) ProposerSettings(_ context.Context) (*validatorServiceConfig.ProposerSettings, error) {
```

Comment:
is it worth using the context here for a trace log?

Reply:
I don't use the context at all for the filesystem DB.
However, context is historically defined in the interface for the BoltDB database. It is effectively used for a trace log.